simapp/app.go

```diff
@@ -569,7 +569,18 @@ func (app *SimApp) GetTKey(storeKey string) *storetypes.TransientStoreKey {
 //
 // NOTE: This is solely used for testing purposes.
 func (app *SimApp) GetMemKey(storeKey string) *storetypes.MemoryStoreKey {
-	return app.memKeys[storeKey]
+	msk := app.memKeys[storeKey]
+	if msk != nil {
+		return msk
+	}
+
+	sk := app.UnsafeFindStoreKey(storeKey)
```

Comment:
When would a mem key be found in the list of real store keys?

Reply:
`provideMemoryStoreKey` calls `registerStoreKey` which appends the key to the list of store keys. Should we have a separate list for them?

https://github.com/cosmos/cosmos-sdk/blob/8eaff8fadd408762fa5a3c5a40969bda4548057e/runtime/module.go#L115-L119